<launch>
  <arg name="first_tb3_prefix"  default="tb3_0"/>
  <arg name="first_tb3_model" default="burger_5"/>
  <arg name="first_tb3_x" default=" 0.0"/>
  <arg name="first_tb3_y" default=" 0.0"/>
  <arg name="first_tb3_z" default=" 0.0"/>
  <arg name="first_tb3_yaw"   default=" 0.0"/>
  <arg name="first_tb3_id"    default=" 5"/>
  
  <arg name="charuco_start_x"    default=" 2.0"/>
  <arg name="charuco_start_y"    default=" 0.0"/>
  <arg name="charuco_start_yaw"  default=" 0.0"/>

  <!-- Launch gazebo world -->
  <arg name="debug" default="false" />
  <arg name="gui" default="true" />
  <arg name="pause" default="false" />
  <arg name="world" default="$(find dse_simulation)/world/charuco.world" />
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(arg world)" />
    <arg name="debug" value="$(arg debug)" />
    <arg name="gui" value="$(arg gui)" />
    <arg name="paused" value="$(arg pause)" />
    <arg name="use_sim_time" value="true" />
  </include>

  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find dse_simulation)/rviz/3_agents_4_tags_dse_odom_config.rviz" />

  <param name="robot_description" command="$(find xacro)/xacro --inorder $(find dse_turtlebot_descriptions)/urdf/turtlebot3_$(arg first_tb3_model).urdf.xacro" />

  <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
    <param name="publish_frequency" type="double" value="50.0" />
  </node>

  <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-urdf -model $(arg first_tb3_prefix) -x $(arg first_tb3_x) -y $(arg first_tb3_y) -z $(arg first_tb3_z) -Y $(arg first_tb3_yaw) -param robot_description" />

  <node pkg="dse_simulation" type="charuco_calibrate_camera.py" name="charuco_calibration_node" output="screen"/>
</launch>
